app/uk/gov/hmrc/eoricommoncomponent/frontend/services/ApplicationService.scala

```diff
@@ -35,65 +40,60 @@ class ApplicationService @Inject() (
 )(implicit ec: ExecutionContext)
     extends EnrolmentExtractor {
 
-  private def isEnrolmentInUse(service: Service, groupId: String, loggedInUser: LoggedInUserWithEnrolments)(implicit
-    hc: HeaderCarrier
-  ): Future[Option[ExistingEori]] =
-    groupEnrolment.groupIdEnrolments(groupId).flatMap { groupEnrolments =>
-      existingEoriForUserOrGroup(loggedInUser, groupEnrolments) match {
-        case Some(existingEori) => enrolmentStoreProxyService.isEnrolmentInUse(service, existingEori)
-        case _                  => Future.successful(None)
-      }
+  private def getEnrolmentInUse(
+    service: Service,
+    groupEnrolments: List[EnrolmentResponse],
+    loggedInUser: LoggedInUserWithEnrolments
+  )(implicit hc: HeaderCarrier): Future[Option[ExistingEori]] =
+    existingEoriForUserOrGroup(loggedInUser, groupEnrolments) match {
+      case Some(existingEori) => enrolmentStoreProxyService.isEnrolmentInUse(service, existingEori)
+      case _                  => Future.successful(None)
     }
 
-  private def isUserEnrolledFor(loggedInUser: LoggedInUserWithEnrolments, service: Service): Boolean =
-    enrolledForService(loggedInUser, service).isDefined
-
-  private def isUserEnrolledForOtherServices(loggedInUser: LoggedInUserWithEnrolments): Boolean =
+  private def isUserEnrolledToOtherService(loggedInUser: LoggedInUserWithEnrolments) =
     enrolledForOtherServices(loggedInUser).isDefined
 
-  private def checkAllServiceEnrolments(loggedInUser: LoggedInUserWithEnrolments, groupId: String)(implicit
-    hc: HeaderCarrier,
-    request: Request[_]
-  ): Future[Either[JourneyError, JourneyType]] =
-    if (isUserEnrolledForOtherServices(loggedInUser))
-      Future.successful(Right(AutoEnrolment))
-    else
-      groupEnrolment.checkAllServiceEnrolments(groupId).flatMap {
-        case Some(groupEnrolment) if groupEnrolment.eori.isDefined =>
-          cache.saveGroupEnrolment(groupEnrolment).map(_ => Right(AutoEnrolment))
-        case _ =>
-          Future.successful(Right(LongJourney))
-      }
+  private def isGroupEnrolledToOtherService(groupEnrolments: List[EnrolmentResponse])(implicit request: Request[_]) = {
+    def checkSupportedServiceEnrolments(groupEnrolments: List[EnrolmentResponse]) = {
+      val serviceList = Service.supportedServicesMap.values.toList
+      groupEnrolments.find(enrolment => serviceList.contains(enrolment.service))
```

Comment:
This internal method seems a little redundant IMO, if you make ServiceList a callable val in the Service object, that could be useful elsewhere. Maybe a lazy val?